### ISSUE TITLE
Paper Translations

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaDiffEq/NeuralNetDiffEq.jl#1]
Lagaris, Isaac E., Aristidis Likas, and Dimitrios I. Fotiadis. "Artificial neural networks for solving ordinary and partial differential equations." IEEE Transactions on Neural Networks 9, no. 5 (1998): 987-1000.

Just to log some details;

$$N(x)=\sum_{i=1}^{i=H} v_i \sigma(\sum_{j=1}^{j=N} w_{ij} x_j +u_j) + k_i$$

> So in the network defined: 
 - $$v_i = $$ elements of `P[3]` 
 - $$ w_{ij} = $$ elements of `P[1]`
 - $$u_i = $$ elements of `P[2]` 
- $$k_i = $$ elements of `P[4]`
- `P[4]` is not in the paper, but doesn't effect the derivative

**Comment by ChrisRackauckas** [comment]
$$\frac{\partial N}{\partial x} = \sum_i v_i w_{ij} \sigma^{\prime} (\sum_j^N w_{ij} x_j + u_j) $$

$$ E[p] = \sum_i ( N(x_i,p) + x_i \frac{\partial N}{\partial x} - f(x_i , A + x_i N(x_i,p))^2 $$

**Comment by ChrisRackauckas** [comment]
@oxinabox w.r.t. `P[4]`

>It is one more parameter to learn.
And since it doesn't show up in the loss, it will be impossible to learn.

**Comment by oxinabox** [comment]
URL: https://arxiv.org/pdf/physics/9705023.pdf

**Comment by edwinksl** [comment]
Hi all, I randomly came across this project from GSOC's page at https://summerofcode.withgoogle.com/projects/#5850956641075200. For my class project for MIT's 6.867 Machine Learning class (http://student.mit.edu/catalog/search.cgi?search=6.867&style=verbatim), I had formulated a feedforward neural network with an arbitrary number of hidden layers to solve ODEs and PDEs and implemented it to solve an ODE for up to 3 hidden layers. This is in contrast to the equations presented in the Lagaris et al.'s paper where they presented results for 1 hidden layer. Unfortunately, I wrote my code in MATLAB, so it is not directly usable for this project. Nonetheless, if you are interested in using a deep neural net for solving ODEs and PDEs, I am more than happy to share my project writeup and code if you think they may be helpful.

**Comment by ChrisRackauckas** [comment]
Hey, thanks for the interest!

Yes, we are using deep neural nets with user-chosen numbers of hidden layers. However, we are still seeing issues with the setup trying to converge as `dt->0`. You might also want to check out TensorFlowDiffEq.jl. I would love to take a look through your project writeup and see what you came up with!

**Comment by edwinksl** [comment]
@ChrisRackauckas Sounds good, let me clean it up a little and I will provide a link to the writeup and code.

**Comment by edwinksl** [comment]
Here is a Dropbox link to the PDF of my writeup: https://www.dropbox.com/s/3ns5ui3lbe754es/Report.pdf?dl=0. I plan to get a repo up on GitHub with the MATLAB and LaTeX code as soon as I can.

**Comment by ChrisRackauckas** [comment]
Thanks for sharing! I see from your Figure 1 that you had the convergence rate issues. Did you have any tests for `error->0` as `N->infty`?

**Comment by edwinksl** [comment]
@ChrisRackauckas Nope, I didn't systemically increase `N` to see if `error` goes to `0`. On hindsight, I should have done that, but I didn't really have much time to do this project haha.

By gradient calculations, I guess you are referring to the appendix? I do think the appendix may be of most interest to you too since the math gets really messy when we try to backpropagate second and higher order derivatives with more than 1 hidden layer. Would be nice to have more eyes on it to make sure the math checks out.
